jerry-core/include/jerryscript-port.h

```diff
@@ -138,6 +138,54 @@ double jerry_port_get_current_time (void);
  */
 struct jerry_instance_t *jerry_port_get_current_instance (void);
 
+/*
+ * Debugger Port API
+ */
+
+/**
+ * A forward declaration of various debugger structures.
+ */
+typedef struct jerry_debugger_transport_t jerry_debugger_transport_t;
+typedef struct jerry_debugger_config_t jerry_debugger_config_t;
+typedef struct jerry_debugger_conn_t jerry_debugger_conn_t;
```

Comment:
Why do we need to forward declare these functions?

Reply:
these needs to be forward declared because it's up to the jerry_port layers to provide the definitions for these, it will run into compile errors if debugger is not enabled.

Reply:
The struct definitions itself is up to the implementation of the transport. Initially, @jimmy-huang had used `void *`, but I suggested to use forward declarations. That way the compiler will still help with ensuring the correct types are used.